### ISSUE TITLE
G4 Doorbell - Ring Notification

### ISSUE THREAD

**Issue by LeSascha** [hjdhjd/homebridge-unifi-protect#1023]
### Homebridge UniFi Protect Version

v6.16.0

### Homebridge Platform and OS

Ubuntu Focal Fossa (20.04.6 LTS)

### Homebridge Version

v1.7.0

### Node Version

v10.19.0

### UniFi OS Version

UniFi OS 3.1.16

### UniFi Protect Controller Version

2.8.35

### Describe the problem

Hi,
I tried to setup this plugin and it's working great so far.
I only have one problem:

- It does not push a notification to my devices (iPhone, ATV, MacBook, ...) when somebody rings on my doorbell

I see the ring gets detected in the Homebridge logs and the Original Unifi Protect Apps sends an notification to my smartphone.

Could someone help me? Probably somethings wrong with my configuration

Thanks in advance 
Sascha

### Homebridge HBUP JSON configuration

```json
{
    "bridge": {
        "name": "Homebridge 2560",
        "username": "******",
        "port": 51239,
        "pin": "******",
        "advertiser": "bonjour-hap"
    },
    "accessories": [],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        },
        {
            "controllers": [
                {
                    "address": "******",
                    "username": "******",
                    "password": "******"
                }
            ],
            "name": "UniFi Protect",
            "platform": "UniFi Protect",
            "options": [
                "Enable.Video.Stream.Only.Low.68D79ADFE915",
                "Enable.Doorbell.Trigger.D021F97502D9",
                "Enable.Doorbell.Trigger.68D79ADFE915"
            ],
            "_bridge": {
                "username": "******",
                "port": 54885
            }
        }
    ]
}
```

### Relevant log output

```shell
[11/17/2023, 1:10:17 AM] [Homebridge UI] Running job to cleanup config.json backup files older than 60 days...
[11/17/2023, 1:23:39 AM] [Homebridge UI] Running scheduled instance backup...
[11/17/2023, 1:23:39 AM] [Homebridge UI] Creating temporary backup archive at /tmp/homebridge-backup-CZoyWr/homebridge-backup-0EE6F5EA2560.tar.gz
[11/17/2023, 12:34:42 PM] [Homebridge UI] [sascha] Accessory layout changes saved.
[11/17/2023, 12:34:46 PM] [Homebridge UI] [sascha] Accessory layout changes saved.
[11/17/2023, 12:34:53 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getDevices
[11/17/2023, 12:34:53 PM] [Homebridge UI] [homebridge-unifi-protect] (node:650103) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
(Use `node --trace-deprecation ...` to show where the warning was created)
[11/17/2023, 12:34:54 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:35:36 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:35:44 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getDevices
[11/17/2023, 12:35:44 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:36:11 PM] [Homebridge UI] Changes to config.json saved.
[11/17/2023, 12:36:11 PM] [Homebridge UI] [homebridge-unifi-protect] Terminating child process...
[11/17/2023, 12:36:11 PM] [Homebridge UI] [homebridge-unifi-protect] Child process ended
[11/17/2023, 12:36:14 PM] [UniFi Protect] Restarting child bridge...
[11/17/2023, 12:36:14 PM] Got SIGTERM, shutting down child bridge process...
[11/17/2023, 12:36:19 PM] [UniFi Protect] Child bridge process ended
[11/17/2023, 12:36:19 PM] [UniFi Protect] Process Ended. Code: 143, Signal: null
[11/17/2023, 12:36:26 PM] [UniFi Protect] Restarting Process...
[11/17/2023, 12:36:27 PM] [UniFi Protect] Launched child bridge with PID 650141
[11/17/2023, 12:36:27 PM] Registering platform 'homebridge-unifi-protect.UniFi Protect'
[11/17/2023, 12:36:27 PM] [UniFi Protect] Loaded homebridge-unifi-protect v6.16.0 child bridge successfully
[11/17/2023, 12:36:27 PM] Loaded 2 cached accessories from cachedAccessories.0EB36EE4785B.
[11/17/2023, 12:36:27 PM] Homebridge v1.7.0 (HAP v0.11.1) (UniFi Protect) is running on port 54885.
[11/17/2023, 12:36:27 PM] [UniFi Protect] Using FFmpeg version: 6.0-homebridge-alpine-x86_64-static.
(node:650141) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
(Use `node --trace-deprecation ...` to show where the warning was created)
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS]: Connected to 192.168.1.9 (UniFi Protect 2.8.35 running on UniFi OS 3.1.16).
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS]: Discovered nvr: Cloud Key [UCK G2 Plus] (address: 192.168.1.9 mac: D021F97502D9).
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS]: Discovered camera: G4 Doorbell [G4 Doorbell] (address: 192.168.1.29 mac: 68D79ADFE915).
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS]: Discovered chime: UP Chime [UP Chime] (address: 192.168.1.31 mac: E4388350D69F).
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Video streaming configured to use only: Low.
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Enabling doorbell automation trigger.
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Enabled doorbell message switch: LEAVE PACKAGE AT DOOR.
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Enabled doorbell message switch: DO NOT DISTURB.
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Enabled doorbell message switch: SASCHA LEBER.
[11/17/2023, 12:36:28 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Doorbell message set: SASCHA LEBER.
[11/17/2023, 12:36:53 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Streaming request from 192.168.1.33: 640x480@30fps, 164 kbps. Using 480x360@15fps (Low), 200 kbps.
[11/17/2023, 12:36:59 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Stopped video streaming session.
[11/17/2023, 12:37:47 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getDevices
[11/17/2023, 12:37:47 PM] [Homebridge UI] [homebridge-unifi-protect] (node:650200) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
(Use `node --trace-deprecation ...` to show where the warning was created)
[11/17/2023, 12:37:48 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:38:18 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:38:23 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:38:24 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:38:32 PM] [Homebridge UI] [homebridge-unifi-protect] Incoming Request: /getOptions
[11/17/2023, 12:40:01 PM] [Homebridge UI] [homebridge-unifi-protect] Terminating child process...
[11/17/2023, 12:40:01 PM] [Homebridge UI] [homebridge-unifi-protect] Child process ended
[11/17/2023, 12:47:32 PM] [UniFi Protect] Cloud Key [UCK-G2-PLUS] G4 Doorbell [G4 Doorbell]: Doorbell ring detected.
```

### Acknowledgment that you are only running UniFi OS and UniFi Protect releases from the Ubiquiti Official release channel

- [X] I acknowledge that I am not running any early access / beta firmwares or operating systems from either Ubiquiti or Apple.

**Comment by zoic21** [comment]
Hello,

I think I have same issue for first ring after homebridge restart. After that first ring no problem for next ring. I have also a g4 doorbell

**Comment by hjdhjd** [comment]
@zoic21 Please don’t hijack another user’s support thread. If you’d like support, please open your own support request. You’re welcome to monitor this one, but I’m only responding to and am going to address the original issue opener.

**Comment by hjdhjd** [comment]
@LeSascha The logs you shared show that the doorbell ring is detected and passed on to HomeKit by HBUP. Beyond that, HBUP has zero control over what HomeKit chooses to do. It’s not atypical when there are new iOS major releases that doorbell rings become flakey for a bit for native and non-native HomeKit doorbells.

TL;DR: nothing I can do here…HBUP is doing what it should, this is a HomeKit in your home issue. You can try rebooting your HomeKit hubs and going through the usual things people try when HomeKit gets a little befuddled. You’re also welcome to try the Discord and see if the community has additional suggestions.

Not satisfying…but unfortunately HomeKit’s a bit of a black box and beyond informing it of the doorbell ring, which we’ve done here, the rest is up to HomeKit.
